### DISCUSSION TITLE
JS dropdown list help

### DISCUSSION THREAD

**Discussion by Mattress1975** [tinymce/tinymce#9976] [General]
Probably a very easy question but I'm pretty new with JS.   I'm trying to create a dropdown list for a web application that once a value is selected it can no longer be picked.  This is going to be for a  baseball lineup creator so the dropdown would be a list of 9 names and there would be 9 positions to fill.  So if a player is selected for the pitcher position their name would be placed in a table next to the position, but when you go to the next position the dropdown wouldn't include the selected players name.   Any help is appreciated and please keep in mind I am a beginner.

**Comment by lostkeys** [comment]
Hi @Mattress1975! An easy answer, you won't be able to build this with any of the TinyMCE menu API's. To offer some unsolicited advice, since TinyMCE is a text editor, it won't stop anyone from simply typing the name, so make sure you're not overthinking the problem. 

Another approach could be to use a [URL dialog](https://www.tiny.cloud/docs/tinymce/latest/urldialog/) which is basically a dialog with an iframe. With that you can build your own mini app and put into the dialog to tick off names as they are inserted. 

Yet another approach could be to [check out our mentions plugin](https://www.tiny.cloud/docs/tinymce/latest/mentions/) where you can build logic to keep track of the names you insert and remove them from the mentions fetch callback.

— Fredrik, Product Manager
